dll/win32/browseui/shellfind/CSearchBar.cpp

```diff
@@ -131,6 +137,34 @@ LRESULT CSearchBar::OnSearchButtonClicked(WORD wNotifyCode, WORD wID, HWND hWndC
         return 0;
     }
 
+    // See if we have a szFileName, our first character is not an asterisk,
+    // and we have room in MAX_PATH to add an asterisk at the beginning of szFileName then add it
+    if(wcscmp(pSearchStart->szFileName, L"") != 0)
+        {
+        len = wcslen(pSearchStart->szFileName);
+        if(len > 0)
+            {
+            ptrchar = pSearchStart->szFileName[len - 1];
+            memcpy(endchar, &ptrchar, 2);
+            ptrchar = pSearchStart->szFileName[0];
+            memcpy(startchar, &ptrchar, 2);
+            if ((len < MAX_PATH) && (wcscmp(startchar, L"*") != 0))
+                {
+                    memmove(&pSearchStart->szFileName[1], &pSearchStart->szFileName[0],
+                            len * sizeof(WCHAR) + sizeof(WCHAR));
+                    len = len + 1;
+                    pSearchStart->szFileName[0] = asterisk[0];
+                }
+            }
+
+        // See if our last character is an asterisk and if not and we have room then add one
+        if((wcscmp(endchar, L"*") != 0) && (len < MAX_PATH))
+            wcscat(pSearchStart->szFileName, L"*");
```

Comment:
It may be nice using the StringSafe functions there.

Reply:
Since we are in control of the size of the asterisk, my understanding is that the comparison stops when the shortest string runs out of characters and the wcscat is only adding a single character which we control and we verified that len < MAX_PATH, so we have room for this extra character. Therefore we are assured that there will not be a problem here, right?

Reply:
You're right that sufficient checks make `wcscat` and similar functions safe. It is still best practice to _never_ use them, because sometimes there are bugs in those checks.

Reply:
OK. Changes made. Thanks.